spec/basic_spec.rb

```diff
@@ -2,56 +2,168 @@
-    subject do
-      CommaSplice::FileCorrector.new(test_csv_path('unescaped-commas-and-non-csv-header.csv'))
-    end
+  context 'with no separator specified' do
+    context 'unescaped-commas-and-non-header' do
+      subject do
+        CommaSplice::FileCorrector.new(test_csv_path('unescaped-commas-and-non-csv-header.csv'))
+      end
 
-    it 'should make proper corrections' do
-      fixed_contents = read_test_csv('unescaped-commas-and-non-csv-header-fixed.csv')
-      subject.save('test-file.txt')
+      it 'should make proper corrections' do
+        fixed_contents = read_test_csv('unescaped-commas-and-non-csv-header-fixed.csv')
+        subject.save('test-file.txt')
 
-      expect(File.read('test-file.txt')).to eq(fixed_contents)
+        expect(File.read('test-file.txt')).to eq(fixed_contents)
+      end
     end
-  end
 
-  context 'unescaped-commas' do
-    subject do
-      CommaSplice::FileCorrector.new(test_csv_path('unescaped-commas.csv'))
-    end
+    context 'unescaped-commas' do
+      subject do
+        CommaSplice::FileCorrector.new(test_csv_path('unescaped-commas.csv'))
+      end
 
-    it 'should make proper corrections' do
-      fixed_contents = read_test_csv('unescaped-commas-fixed.csv')
-      subject.save('test-file.txt')
+      it 'should make proper corrections' do
+        fixed_contents = read_test_csv('unescaped-commas-fixed.csv')
+        subject.save('test-file.txt')
 
-      expect(File.read('test-file.txt')).to eq(fixed_contents)
+        expect(File.read('test-file.txt')).to eq(fixed_contents)
+      end
     end
-  end
 
-  context 'unescaped-quotes' do
-    subject do
-      CommaSplice::FileCorrector.new(test_csv_path('unescaped-quotes.csv'))
+    context 'unescaped-quotes' do
+      subject do
+        CommaSplice::FileCorrector.new(test_csv_path('unescaped-quotes.csv'))
+      end
+
+      it 'should make proper corrections' do
+        fixed_contents = read_test_csv('unescaped-quotes-fixed.csv')
+        subject.save('test-file.txt')
+
+        expect(File.read('test-file.txt')).to eq(fixed_contents)
+      end
     end
 
-    it 'should make proper corrections' do
-      fixed_contents = read_test_csv('unescaped-quotes-fixed.csv')
-      subject.save('test-file.txt')
+    context 'unknown combo should prompt for option' do
+      subject do
+        CommaSplice::FileCorrector.new(test_csv_path('10000-maniacs.csv'))
+      end
 
-      expect(File.read('test-file.txt')).to eq(fixed_contents)
+      it 'should prompt for correction' do
+        expect($stdin).to receive(:gets).and_return('4')
+
+        fixed_contents = read_test_csv('10000-maniacs-fixed.csv')
+        subject.save('test-file.txt')
+        expect(File.read('test-file.txt')).to eq(fixed_contents)
+      end
     end
-  end
 
-  context 'unknown combo should prompt for option' do
-    subject do
-      CommaSplice::FileCorrector.new(test_csv_path('10000-maniacs.csv'))
+    context 'with comma as separator' do
```

Comment:
I think for the new separator arguments, you could instead write one of the tests into the new 'with no separator specified' context and then replicate that one test across all three contexts.  There's no reason to believe that if the basic functionality works with arbitrary delimiters then 'unknown combo should prompt for option' would break, for instance.  The downside of this approach is that it triples the test suite run time.  Maybe a new test could be written instead to avoid having multiple versions of the fixture file.

Reply:
I'll change the code. Please check if the new version and let me know